meta-resin-common/recipes-support/resin-hostname/resin-hostname/resin-hostname

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+set -e
+
+. /usr/sbin/resin-vars
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    NEW_HOSTNAME=$(jq -r '.uuid //empty' $CONFIG_PATH)
+fi
+
+if [ -z "$NEW_HOSTNAME" ]; then
+    echo "[ERROR]: Unexpected uuid missing from config.json."
+    exit 1
+fi
+
+# Set hostname temporary hostname
+OLD_HOSTNAME=$(hostname)
+hostname "$NEW_HOSTNAME"
+
+# Commit the new hostname
+echo "$NEW_HOSTNAME" > /etc/hostname
```

Comment:
If we have `hostnamectl` available in our host (it's a systemd thing) it'd better if we used it https://www.freedesktop.org/software/systemd/man/hostnamectl.html